rfc/private-key-handling.md

```diff
@@ -0,0 +1,173 @@
+# RFC: Standardizing Private Key Material Handling
+
+## Summary
+
+This RFC proposes standardizing how we handle private key material across our services, adopting PEM format as the single standard for private key storage and configuration.
+
+## Problem Statement
+
+Our services currently accept private keys in multiple formats:
+
+- PEM files (with inconsistent encryption practices).
+- JSON files (non-standard, service-specific schemas).
+- Base64 encoded strings (raw key material, no metadata).
+- Multibase base64 padded strings (uncommon in standard tooling).
+
+This creates:
+
+- **Security inconsistency**: Different formats have varying security properties.
+- **Implementation complexity**: Each service implements multiple parsers.
+- **Cognitive overhead**: Developers must remember which format works where.
+- **Error-prone deployments**: Format mismatches cause configuration failures.
+
+## Proposed Solution: PEM Standard
+
+**Format**: PEM-encoded private keys using PKCS#8 format
+
+```
+-----BEGIN PRIVATE KEY-----
+MC4CAQA...
+-----END PRIVATE KEY-----
+```
+
+### Why PEM?
+
+1. **Universal tooling support**: OpenSSL, ssh-keygen, and all major crypto libraries support PEM natively.
+2. **Self-documenting format**: Clear headers (`-----BEGIN PRIVATE KEY-----`) identify content type.
```

Comment:
🤷 the multibase encoded string is also self describing - the decoded bytes are also prefixed with the alogorithm. PEM has the advantage of being human readable to some extent.